polysemy-plugin/src/Polysemy/Plugin/Fundep.hs

```diff
@@ -202,8 +215,9 @@ solveFundep (ref, stuff) given _ wanted = do
     let r  = fcRow fc
     case findMatchingEffectIfSingular fc given_finds of
       -- We found a real given, therefore we are in the context of a function
-      -- with an explicit @Member e r@ constraint.
-      Just eff' -> mkWanted fc FunctionDef eff'
+      -- with an explicit @Member e r@ constraint. We also know it can
+      -- be unified (although it may generate unsatisfiable constraints).
```

Comment:
What do those unsatisfiable constraints look like? Can we add some tests for multiple states and see what happens in that case? I'm particularly curious about two concrete `Num`s.

Reply:
What I mean by unsatisfiable constraints in this context is `Num String`; in particular, I don't mean obviously unsatisfiable constraints like `String ~ Int`.

For example, `polysemy-plugin` will generate a `Num String` constraint for code like this:

```haskell
ex :: Member (State String) r => Sem r ()
ex = put 10
```

With or without my modification, because it sees the type variable created by the literal, and `canUnifyRecursive` says "oh hey, we can unify string with that".

All my modification does is use `canUnifyRecursive` to get rid of cases we know unification won't work out, even without knowing what constraints we are given.

I'm not sure what you mean about two concrete `Num`s.